lib/modules/platform/github/index.ts

```diff
@@ -759,26 +762,57 @@ export async function findPr({
 
 const REOPEN_THRESHOLD_MILLIS = 1000 * 60 * 60 * 24 * 7;
 
+async function remoteBranchExists(
+  repo: string,
+  branchName: string
+): Promise<boolean> {
+  const RefSchema = z
+    .object({
+      ref: z.string().transform((val) => val.replace(/^refs\/heads\//, '')),
+    })
+    .transform(({ ref }) => ref);
+
+  const { val, err } = await githubApi
+    .getJsonSafe(
```

Comment:
there is no other way to get this info with a head request? 😕 (asking because of rate limits, head requests don't count)

Reply:
Ha, I think I've found the way

Reply:
```
❯ xh head https://api.github.com/repos/zharinov/renovate/git/refs/heads/foo                
HTTP/2.0 200 OK
...

❯ xh head https://api.github.com/repos/zharinov/renovate/git/refs/heads/foo/
HTTP/2.0 200 OK
...

❯ xh head https://api.github.com/repos/zharinov/renovate/git/refs/heads/foo/bar
HTTP/2.0 200 OK
...

❯ xh head https://api.github.com/repos/zharinov/renovate/git/refs/heads/foo/bar/
HTTP/2.0 404 Not Found
...
```